policy-test/tests/api.rs

```diff
@@ -283,6 +283,221 @@ async fn server_with_authorization_policy() {
+            .expect("watch must return an initial config");
+        tracing::trace!(?config);
+        assert_is_default_all_unauthenticated!(config);
+        assert_protocol_detect!(config);
+
+        // Create a server that selects the pod's proxy admin server and ensure
+        // that the update now uses this server, which has no authorizations
+        // and no routes.
+        let _server = create(&client, mk_admin_server(&ns, "linkerd-admin")).await;
+        let config = rx
+            .next()
+            .await
+            .expect("watch must not fail")
+            .expect("watch must return an updated config");
+        tracing::trace!(?config);
+        assert_eq!(
+            config.protocol,
+            Some(grpc::inbound::ProxyProtocol {
+                kind: Some(grpc::inbound::proxy_protocol::Kind::Http1(
+                    grpc::inbound::proxy_protocol::Http1::default()
+                )),
+            }),
+        );
+        assert_eq!(config.authorizations, vec![]);
+        assert_eq!(
+            config.labels,
+            convert_args!(hashmap!(
+                "group" => "policy.linkerd.io",
+                "kind" => "server",
+                "name" => "linkerd-admin"
+            )),
+        );
+
+        // Create an http route that refers to the `linkerd-admin` server (by
+        // name) and ensure that the update now reflects this route.
+        create(
+            &client,
+            k8s_gateway_api::HttpRoute {
+                metadata: kube::api::ObjectMeta {
+                    namespace: Some(ns.clone()),
+                    name: Some("metrics-route".to_string()),
+                    ..Default::default()
+                },
+                spec: k8s_gateway_api::HttpRouteSpec {
+                    inner: k8s_gateway_api::CommonRouteSpec {
+                        parent_refs: Some(vec![k8s_gateway_api::ParentReference {
+                            group: Some("policy.linkerd.io".to_string()),
+                            kind: Some("Server".to_string()),
+                            namespace: None,
+                            name: "linkerd-admin".to_string(),
+                            section_name: None,
+                            port: None,
+                        }]),
+                    },
+                    hostnames: None,
+                    rules: Some(vec![k8s_gateway_api::HttpRouteRule {
+                        matches: Some(vec![k8s_gateway_api::HttpRouteMatch {
+                            path: Some(k8s_gateway_api::HttpPathMatch::Exact {
+                                value: "/metrics".to_string(),
+                            }),
+                            headers: None,
+                            query_params: None,
+                            method: Some("GET".to_string()),
+                        }]),
+                        filters: None,
+                        backend_refs: None,
+                    }]),
+                },
```

Comment:
take it or leave it: could use `Default` here:
```suggestion
                    inner: k8s_gateway_api::CommonRouteSpec {
                        parent_refs: Some(vec![k8s_gateway_api::ParentReference {
                            group: Some("policy.linkerd.io".to_string()),
                            kind: Some("Server".to_string()),
                            name: "linkerd-admin".to_string(),
                            ..Default::default()
                        }]),
                    },
                    hostnames: None,
                    rules: Some(vec![k8s_gateway_api::HttpRouteRule {
                        matches: Some(vec![k8s_gateway_api::HttpRouteMatch {
                            path: Some(k8s_gateway_api::HttpPathMatch::Exact {
                                value: "/metrics".to_string(),
                            }),
                            method: Some("GET".to_string()),
                            ..Default::default()
                        }]),
                        ..Default::default()
                    }]),
                },
```